src/logic_utils/truth_table.cpp

```diff
@@ -0,0 +1,177 @@
+#include <iostream>
+
+#include <string>
+
+#include <stack>
+
+#include <cctype>
+
+#include <unordered_map>
+#include <map>
+#include <vector>
+
+#include <set>
+
+using namespace std;
+
+
+map < int, char > variables;
+
+
+
+bool isOperator(char c) {
+return c == '+' || c == '*' || c == '~' || c == '!';
+}
+
+int getPrecedence(char op) {
+  if (op == '+')
+    return 1;
+  if (op == '*')
+    return 2;
+  return 0;
+}
+
+double applyOperation(double a, double b, char op) {
+  switch (op) {
+  case '+':
+    return a || b;
+  case '*':
+    return a && b;
+  default:
+    return 0.0; // Invalid operator
+  }
+}
+
+double evaluateExpression(const std::string & expression) {
+  std::stack < double > values;
+  std::stack < char > operators;
+
+  for (size_t i = 0; i < expression.length(); ++i) {
+    char c = expression[i];
```

Comment:
Maybe renaming it to `current_char` would be better since just `c` was confusing at first.